File: 백준/Bronze/1773. 폭죽쇼/폭죽쇼.py
import sys
input = sys.stdin.readline

n, c = map(int, input().split())
num = list(int(input()) for _ in range(n))
time = [0] * (c + 1)
for i in num:
    for j in range(1, c + 1):
        if j % i == 0:
            time[j] = 1
print(sum(time))

File: 백준/Bronze/1773. 폭죽쇼/폭죽쇼.java
import java.util.*;
import java.io.*;

public class Main {

    public static void main(String[] args) throws IOException{
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        String[] input = br.readLine().split(" ");
        int n = Integer.parseInt(input[0]);
        int c = Integer.parseInt(input[1]);
        int[] time = new int[n];
        for (int i=0; i < n; i++) {
            time[i] = Integer.parseInt(br.readLine());
        }

        int cnt = 0;
        for (int i=1; i < c + 1; i++) {
            boolean temp = false;
            for (int j:time) {
                if (i % j == 0) {
                    temp = true;
                }
            }
            if (temp) {
                cnt += 1;
            }
        }
        System.out.println(cnt);
    }
}